### ISSUE TITLE
tests/kernel/arm_irq_vector_table: test case cannot quit displaying "isr 0 ran!"

### ISSUE THREAD

**Issue by cinlyooi-intel** [zephyrproject-rtos/zephyr#13823]
**Describe the bug**
Test case did not quit. It kept displaying "isr 0 ran!" until it times out

**To Reproduce**
1. mkdir build; cd build
2. cmake -DBOARD=nrf52_pca10040 -DARCH=arm 
3. make
4. make run

**Screenshots or console output**
```
***** delaying boot 1000ms (per build configuration) *****
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
...
isr0 ran!
isr0 ran!
isr0 ran!
isr0 ran!
isr0
```

**Environment (please complete the following information):**
 - OS: fedora29
 - Toolchain : zephyr
 - Commit SHA or Version used: dd5ff3957ceece428e3465dd4e23fb926f61691c

**Comment by ioannisg** [comment]
Thanks for reporting this, @cinlyooi-intel 
The problem is that the IRQ line-0 is wired to the CLOCK_CONTROL IRQ in nRF52x series of SOCs, and that peripheral is generating interrupts.

**Comment by ioannisg** [comment]
Provided a fix in #13903, @cinlyooi-intel pls, test if you have time.
